src/ejabberd_local.erl

```diff
@@ -54,15 +54,15 @@
 
 %% Hooks callbacks
 
--export([disco_local_features/1]).
+-export([disco_local_features/3]).
 
 %% gen_server callbacks
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
          terminate/2, code_change/3]).
 
 -export([do_route/4]).
 
--ignore_xref([disco_local_features/1, do_route/4, get_iq_callback/1,
+-ignore_xref([disco_local_features/3, do_route/4, get_iq_callback/1,
```

Comment:
Actually, xref can see when you do `fun ?MODULE:disco_local_features/3` that the function is actually used, maybe not called, but given elsewhere who might do something with it, so it's not dead code. So you can actually remove all there `disco_local_features/3` from the xref ignores everywhere 😄